src/vs/editor/contrib/snippet/snippetParser.ts

```diff
@@ -722,13 +741,29 @@ export class SnippetParser {
 						placeholder.appendChild(choice);
 						parent.appendChild(placeholder);
 						return true;
+					} else if (this._accept(TokenType.Forwardslash)) {
+						// ...|/<regex>/<format>/<options>} -> transform
+						if (this._parseTransform(placeholder)) {
+							parent.appendChild(placeholder);
+							return true;
+						}
```

Comment:
Not sure if that's actually supported by TextMate. Do you have a reference?

Reply:
As far as I know, TextMate only supports transformations for the simple case `${1/.../.../...}`. This two here would be an extension to the TextMate behavior. I agree, that the choice case might not be needed, but the nested one would be nice for default values e.g.

Reply:
For Reference: http://manual.macromates.com/en/snippets#transformations